Source/WebCore/platform/graphics/coreimage/FEComponentTransferCoreImageApplier.mm

```diff
@@ -45,17 +47,52 @@
     ASSERT(supportsCoreImageRendering(effect));
 }
 
+static CIKernel* gammaKernel()
+{
+    static NeverDestroyed<RetainPtr<CIKernel>> kernel;
+    static std::once_flag onceFlag;
+
+    std::call_once(onceFlag, [] {
+        NSError *error = nil;
+        NSArray<CIKernel *> *kernels = [CIKernel kernelsWithMetalString:@R"( /* NOLINT */
+            [[stitchable]] float4 gammaFilter(coreimage::sampler src, float ampR, float expR, float offR,
```

Comment:
I'm not sure what the status quo on such things are, but can be this be compiled at build time and bundled with the framework as a binary, rather than being runtime constructed?

Reply:
Yes, we could certainly do that. The cost is quite minimal given this shader's size, but at least we should file a bugzilla to make the build changes to load from a metallib.

Reply:
sute, it can. perhaps we can address it as a follow-up, since there may be more shaders coming